src/Compilers/CSharp/Portable/CodeGen/EmitAddress.cs

```diff
@@ -64,6 +64,13 @@ private LocalDefinition EmitAddress(BoundExpression expression, AddressKind addr
 
                     if (expression.Type.IsValueType)
                     {
+
+                        if (!HasHome(expression, addressKind))
+                        {
+                            // a readonly method is calling a non-readonly method, therefore we need to copy 'this'
```

Comment:
This might need a little more scrutiny--that nothing questionable happens when passing `this` as an argument to a readonly method on another variable for instance.

Reply:
Refreshing myself on when this would happen. In general we disallow passing a writable address pointing to a readonly `this` variable. But in the special case of instance member invocations on structs we will implicitly create a local, writable copy of `this` variable and pass its address instead of giving an error.

Before the readonly members feature the implicit copy could not happen on `this` directly, because it requires `this` to be readonly in one method and not readonly in another method. LocalFieldAccesses were already using the solution of "check HasHome, emit a temp if it fails". That's why we need it here for this specific scenario.